### ISSUE TITLE
Is it possible to have wheels for pytorch 2.1.2 and cuda 11.8?

### ISSUE THREAD

**Issue by AruniRC** [OutofAi/tiny-cuda-nn-wheels#3]
Thanks so much for providing pre-compiled wheels (since tiny-cuda-nn can get difficult to install on some platforms)!

Is it possible to share similar wheels for  `pytorch 2.1.2` and `cuda 11.8`? These dependencies are used by Nerfstudio, and it would really help to have wheels in this case.

Thanks,
Aruni

**Comment by OutofAi** [comment]
Are you planning to run it locally or on Colab, if so which GPU you are using? cause the difficult part with cuda and wheel generation is that the processing unit needs to be clear as well, so you can generate wheel based of the arch type

**Comment by AruniRC** [comment]
Thank you for responding.

Planning to run it locally, and GPUs can be any of Ampere/Ada/Hopper/Turing. I was wondering if having pre-build wheels for any (or all?) of these platforms would be feasible, or perhaps in this case local builds on each architecture is the more hassle-free way to go ..

**Comment by OutofAi** [comment]
sorry, I meant compute compatibility, not necessarily architecture, the values displayed in this https://developer.nvidia.com/cuda-gpus
if it's a handful, you can easily fork the repo and change the file _github tiny-cuda-nn-wheels/.github/workflows/wheels-generator.yml_ and run the generator to get the files you need,  and then run Actions -> CI -> Run Workflow to generate the relevant wheel for you

**Comment by AruniRC** [comment]
Yes, I realize this is a handful and I'm better off generating the wheels. I'll take a look at your generator yaml and try this out.  Thanks again for creating this repo.

**Comment by gaperezsa** [comment]
Hello Aruni, were you able to create said wheel? would it be possible for you to share it?

**Comment by OutofAi** [comment]
sure, but I would need the platform and the architecture type of the graphics card you wanna run it on to do it? you can also do it yourself if you just fork the github repo and change the  https://github.com/OutofAi/tiny-cuda-nn-wheels/blob/main/.github/workflows/wheels-generator.yml based on your needs

**Comment by OutofAi** [comment]
I kicked off a build for `pytorch 2.1.2` and `cuda 11.8`

**Comment by OutofAi** [comment]
@gaperezsa @AruniRC they are now available, or you can just run this script depending on your architecture to download and install it on your machine

```
import os
import subprocess

# Mapping of architecture to post numbers
arch_to_post = {
    'Turing': '75',
    'Ampere+Tegra': '87',
    'Ampere': '80',
    'Ada': '89',
    'Hopper': '90'
}

# Base URL for the wheel files
base_url = "https://github.com/OutofAi/tiny-cuda-nn-wheels/releases/download/1.7.2/"

def download_and_install_wheel(architecture, version='1.7', directory='.'):
    # Get the post number for the architecture
    post = arch_to_post.get(architecture)
    
    if post is None:
        print(f"Error: Unsupported architecture {architecture}")
        return
    
    # Construct the wheel file name
    wheel_filename = f"tinycudann-{version}.post{post}212118-cp310-cp310-linux_x86_64.whl"
    wheel_url = f"{base_url}{wheel_filename}"
    
    # Download the wheel file
    print(f"Downloading {wheel_filename} from {wheel_url}...")
    download_command = f'curl -L "{wheel_url}" -o {wheel_filename}'
    subprocess.run(download_command, shell=True, check=True)
    
    # Check if the file was downloaded successfully
    if os.path.exists(wheel_filename):
        # Install the wheel using pip
        print(f"Installing {wheel_filename} for {architecture} architecture...")
        subprocess.run(['pip', 'install', wheel_filename], check=True)
    else:
        print(f"Error: Failed to download {wheel_filename}")

# Example usage:
download_and_install_wheel('Ada')
import tinycudann as tcnn 
```
